src/draw_target.rs

```diff
@@ -514,7 +514,7 @@ impl DrawState {
                 // subtract with overflow later.
                 usize::max(terminal_len, 1)
             };
-            if real_len + diff >= term_height {
+            if real_len + diff > term_height {
```

Comment:
I'd prefer for fixes like these to be squashed into the originating commit.

(Also goes for the other changes you've tacked on.)